src/tests/user.test.js

```diff
@@ -191,4 +193,112 @@ describe('User Signin', () => {
+        .post('/user/signin')
+        .send(user_credentials)
+        .then((login_response) => {
+          let token = 'Bearer ' + login_response.body.token;
+          chai.request(app).get('/user/').set('Authorization', token)
+              .end((err, res) => {
+                expect(res).to.have.status(200);
+              })
+          }).catch(err => {
+              console.log(err);
+          })
+  });
+})
+
+describe('/GET/:id users', () => {
+
+	it('it should GET a user by the given id', () => {
+    
+    const user_credentials = {
+      "email": "keza@gmail.com",
+      "password": "1234567avb$#8"
+    }
+    let user = new User({firstname: "kalisa", lastname: "wilson", email: "wizo@gmail.com", password: "12345678#$"});
+
+    chai.request(app)
+        .post('/user/signin')
+        .send(user_credentials)
+        .then((login_response) => {
+          let token = 'Bearer ' + login_response.body.token;
+          user.save((err, user) => {
+            chai.request(app)
+                .get('/user/' + user.id)
+                .send(user)
+                .set('Authorization', token)
+                .end((err, res) => {
+                  expect(res).to.have.status(200);
+                })
+            })
+          });
+  });
+
+})
+
+describe('/PUT/:id users', () => {
+
+	it('it should update a user by the given id', () => {
+    
+    const user_credentials = {
+      "email": "keza@gmail.com",
+      "password": "1234567avb$#8"
+    }
+    let user = new User({firstname: "kalisa", lastname: "wilson", email: "wizo@gmail.com", password: "12345678#$"});
+
+    chai.request(app)
+        .post('/user/signin')
+        .send(user_credentials)
+        .then((login_response) => {
+          let token = 'Bearer ' + login_response.body.token;
+          user.save((err, user) => {
+            chai.request(app)
+                .put('/user/' + user.id)
+                .send({firstname: "Mussa", lastname: "wilson"})
+                .set('Authorization', token)
+                .end((err, res) => {
+                  expect(res).to.have.status(200);
+                })
+            })
+          });
+  });
+})
+
+describe('/DELETE/:id users', () => {
+
+	it('it should delete a user by the given id', () => {
+    
+    const user_credentials = {
+      "email": "keza@gmail.com",
+      "password": "1234567avb$#8"
+    }
+    let user = new User({firstname: "Peter", lastname: "wilson", email: "wizo@gmail.com", password: "12345678#$"});
+
+    chai.request(app)
+        .post('/user/signin')
+        .send(user_credentials)
+        .then((login_response) => {
+          let token = 'Bearer ' + login_response.body.token;
+          user.delete((err, user) => {
+            chai.request(app)
+                .get('/user/' + user.id)
+                .set('Authorization', token)
+                .end((err, res) => {
+                  expect(res).to.have.status(200);
+                })
+            })
+          });
+  });
```

Comment:
Here, you should be testing `GET`, `PUT`, `DELETE` endpoints but you're only sending a login request instead. Double check that.